docker-compose.yml

```diff
@@ -45,19 +44,59 @@ services:
       - ${ZTG_CHAIN_RPC_PORT}:9933
       - ${ZTG_CHAIN_WS_PORT}:9944
     networks:
-      - blockchain
+      - backnet
     volumes:
       - ${BASE_PATH}:/zeitgeist/data
     command: --base-path=/zeitgeist/data --dev --ws-external --rpc-external --rpc-cors all --rpc-methods Unsafe
 
+  subsquid-db:
+    container_name: subsquid-db
+    image: postgres:14
+    networks:
+      - backnet
+    environment:
+      POSTGRES_DB: postgres
+      POSTGRES_PASSWORD: postgres
+    volumes:
+      - ${BASE_PATH}/subsquid:/var/lib/postgresql/data
+
+  subsquid:
+    container_name: subsquid
+    image: zeitgeistpm/squid-processor:latest
+    depends_on:
+      - subsquid-db
+      - ipfs
```

Comment:
When starting from a fresh instance, Subsquid will shut down because the DB is not ready (initialized) yet. Starting a second time will work, because the DB was initialized already and is ready for connection immediately after starting the container. This is the error log:

<details>
  <summary>
  Click to expand docker-compose up log
  </summary>
<pre>
<code>
subsquid                | yarn run v1.22.19
subsquid                | $ yarn migration:apply && node lib/main.js
subsquid                | $ squid-typeorm-migration apply
ipfs                    | 2024/01/22 15:25:12 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/quic-go/quic-go/wiki/UDP-Buffer-Sizes for details.
subsquid-db             | syncing data to disk ... ok
subsquid-db             | 
subsquid-db             | 
subsquid-db             | Success. You can now start the database server using:
subsquid-db             | 
subsquid-db             |     pg_ctl -D /var/lib/postgresql/data -l logfile start
subsquid-db             | 
subsquid-db             | initdb: warning: enabling "trust" authentication for local connections
subsquid-db             | You can change this by editing pg_hba.conf or using the option -A, or
subsquid-db             | --auth-local and --auth-host, the next time you run initdb.
subsquid-db             | waiting for server to start....2024-01-22 15:25:12.788 UTC [49] LOG:  starting PostgreSQL 14.10 (Debian 14.10-1.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
subsquid-db             | 2024-01-22 15:25:12.789 UTC [49] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
subsquid-db             | 2024-01-22 15:25:12.799 UTC [50] LOG:  database system was shut down at 2024-01-22 15:25:12 UTC
subsquid-db             | 2024-01-22 15:25:12.804 UTC [49] LOG:  database system is ready to accept connections
subsquid-db             |  done
subsquid-db             | server started
subsquid-db             | 
subsquid-db             | /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
subsquid-db             | 
subsquid-db             | 2024-01-22 15:25:12.899 UTC [49] LOG:  received fast shutdown request
subsquid-db             | waiting for server to shut down....2024-01-22 15:25:12.902 UTC [49] LOG:  aborting any active transactions
subsquid-db             | 2024-01-22 15:25:12.903 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
subsquid-db             | 2024-01-22 15:25:12.903 UTC [51] LOG:  shutting down
subsquid                | Error: connect ECONNREFUSED 172.30.0.2:5432
subsquid                |     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16) {
subsquid                |   errno: -111,
subsquid                |   code: 'ECONNREFUSED',
subsquid                |   syscall: 'connect',
subsquid                |   address: '172.30.0.2',
subsquid                |   port: 5432
subsquid                | }
subsquid-db             | 2024-01-22 15:25:12.962 UTC [49] LOG:  database system is shut down
subsquid                | error Command failed with exit code 1.
subsquid                | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
subsquid                | error Command failed with exit code 1.
subsquid                | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
subsquid-db             |  done
subsquid-db             | server stopped
subsquid-db             | 
subsquid-db             | PostgreSQL init process complete; ready for start up.
subsquid-db             | 
subsquid-db             | 2024-01-22 15:25:13.018 UTC [1] LOG:  starting PostgreSQL 14.10 (Debian 14.10-1.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
subsquid-db             | 2024-01-22 15:25:13.018 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
subsquid-db             | 2024-01-22 15:25:13.018 UTC [1] LOG:  listening on IPv6 address "::", port 5432
subsquid-db             | 2024-01-22 15:25:13.021 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
subsquid-db             | 2024-01-22 15:25:13.054 UTC [62] LOG:  database system was shut down at 2024-01-22 15:25:12 UTC
subsquid-db             | 2024-01-22 15:25:13.059 UTC [1] LOG:  database system is ready to accept connections
</code>
</pre>
</details>

The solution is to integrate a health check into the `subsquid-db` container, that is used as a precondition for the `subsquid` container:
```suggestion
  subsquid-db:
    container_name: subsquid-db
    image: postgres:14
    healthcheck:
      test: /usr/bin/pg_isready
      interval: 5s
      timeout: 10s
      retries: 120
    networks:
      - backnet
    environment:
      POSTGRES_DB: postgres
      POSTGRES_PASSWORD: postgres
    volumes:
      - ${BASE_PATH}/subsquid:/var/lib/postgresql/data

  subsquid:
    container_name: subsquid
    image: zeitgeistpm/squid-processor:latest
    depends_on:
      ipfs:
        condition: service_started
      subsquid-db:
        condition: service_healthy
      zeitgeist-blockchain:
        condition: service_started
```

In addition to the healthcheck, I also added `zeitgeist-blockchain` as a dependency.